### ISSUE TITLE
Proposal: create labels to make issue tracking easier

### ISSUE THREAD

**Issue by lcabrini** [LinuxAccra-UserGroup/la-webapp#1]
I propose that we add a few labels to the issue tracker to make it easier to manage and filter through issues.
1. proposal: this issue for instance should be labelled with the proposal label. It is meant for making a proposal, which can then be discussed and (hopefully) agreed on.
2. junior: simpler tasks should be labelled with this so that less experienced members have an entry-point into the project
3. user story: for tracking user stories, ie. short stories used to convey some sort of functionality.
4. labels for the skills involved, such as php, database, html, css and javascript. This can help somebody who may only know HTML and CSS to filter only tasks relevant to their skill set.

Of course, if you can think of other labels, feel free to suggest them in this thread. Or if you don't like the idea of using labels to manage the issue tracker, argue your case.

**Comment by machelkassah** [comment]
I think labels will be great. It make it easier to follow what's going on.

**Comment by laudb** [comment]
Okay, I'll like to try this.
 I'm also wondering if instead of programming languages for the skills we use: front-end, back-end, etc. especially in cases where there may be many different programming languages, and many different labels?

**Comment by lcabrini** [comment]
@laudb that is also a good idea. The important thing is that we make it easy to find things that we feel we can work on.

**Comment by laudb** [comment]
@lcabrini yeah definitely, I agree. what colors do we give these labels too?

**Comment by lcabrini** [comment]
I think we should stick to the same or at least similar colors for each label category. So, for instance front-end, back-end, etc. should look the same (or similar). What do the rest think?

**Comment by laudb** [comment]
Yeah, this works, but I think proposal should stand out or have a different color as it requires everybody to chip in.

**Comment by lcabrini** [comment]
Yeah, I agree. Basically the idea would be this.
1. proposal: one color, stands out
2. user story: one color, stands out
3. junior: one color, stands out
4. back-end, front-end: one color (or similar colors)
5. state labels, proposed, implemented, tested, accepted, etc.: one color (or similar colors)

I've been reading up a little on what other teams are doing. It seems some teams create labels that have the format type:value, such as state:accepted, state:needs design, level:junior, type:proposal, etc. Is that something we could also investigate?

**Comment by laudb** [comment]
@lcabrini oh okay, I understand now.

yeah, I see this 'type: value' format as well. I think this will give a clearer range of label options. It could even work with colours. E.g. Type: Level -> Value (Junior - White,  Senior -Black) but I think with this we might need to re-colour Github's existing labels to fit our own defined label scheme. 

Perhaps we could start  with:

> proposal: one color, stands out
> user story: one color, stands out
> junior: one color, stands out
> back-end, front-end: one color (or similar colors)
> state labels, proposed, implemented, tested, accepted, etc.: one color (or similar colors)

**Comment by lcabrini** [comment]
Yes, maybe we should. I think at this stage we should just set something up and move on. We can always change it later. No matter how much we discuss, we'll still most likely realized later that we should have done it differently. I am a firm believer that "making mistakes is the best teacher". So let's just set up some labels and start labelling.

Btw, I think github gives the ability to give team members access to setting up labels, milestones etc through access control. If you eventually begin to overwhelmed by tasks such as labelling tasks or assigning issues to milestones, you can count on our help! :)

**Comment by lcabrini** [comment]
Actually, just after I wrote that, I noticed that I can now create and apply both labels and milestones. So, that is cool then.

We probably would still like you to manage it. Just ask us to offload you when it gets too much, that's all.

**Comment by laudb** [comment]
definitely no problem :+1: 

For start, I can take the labels,  @lcabrini can you set up the first milestone for us?
There's more on the table, so all @LinuxAccra-UserGroup/webapp-dev  grab your plates!
